docs/specs/tripgo.swagger.yaml

```diff
@@ -2015,56 +2000,62 @@ definitions:
       - alt
       - localIcon
 
-  SpecificModes:
-    description: Specific modes grouped by type
+  ModeIdentifier:
+    description: This is a map/dictionary with multiple items of this type. Each key is a mode identifier. See https://developer.tripgo.com/faq/#mode-identifiers.
     properties:
-      public_transport:
-        $ref: '#/definitions/PublicTransportModeDetails'
-      taxi_and_tnc:
-        $ref: '#/definitions/TaxiAndTNCModeDetails'
-      shared_vehicles:
-        $ref: '#/definitions/SharedVehiclesModeDetails'
-
-  PublicTransportModeDetails:
-    description: list of public transit services that are accessible to public.
-    allOf:
-    - $ref: '#/definitions/ModeDetails'
-    required:
-      - alt
-      - localIcon
-
-  TaxiAndTNCModeDetails:
-    description: list of taxi-like on-demand services, including regular taxis, Uber, Lyft, etc.
-    allOf:
-    - $ref: '#/definitions/ModeDetails'
-    required:
-      - title
-
-  SharedVehiclesModeDetails:
-    description: list of shared vehicles services, including bikes (like Bicing) and cars (like ZipCar or GoGet)
-    allOf:
-    - $ref: '#/definitions/ModeDetails'
+      title:
+        type: string
+      URL:
+        type: string
+      color:
+        $ref: '#/definitions/Color'
+      icon:
+        type: string
+        description: File name part for downloading an image for this mode. [See separate documentation](TODO) for how to construct a URL from this.
+      implies:
+        type: string
+        description: Another mode identifier that is required by this one, e.g., you can't request only school buses without also requesting regular public transport.
     required:
       - title
 
   ModeDetails:
-    allOf:
-    - $ref: '#/definitions/ModeInfo'
-    - type: object
-      properties:
-        title:
-          type: string
-        URL:
+    description: This is a map/dictionary with multiple items of this type. Each key is a mode identifier. See https://developer.tripgo.com/faq/#mode-identifiers.
+    properties:
+      title:
+        type: string
+      URL:
+        type: string
+      color:
+        $ref: '#/definitions/Color'
+      alt:
+        description: Textual alternative to icon
+        type: string
+      localIcon:
+        description: Part of icon file name that should be shipped with app.
+        type: string
+      remoteIcon:
+        description: Part of icon file name that can be fetched from server.
+        type: string
+      remoteDarkIcon:
+        description: Part of icon file name for dark background that can be  fetched from server.
+        type: string
+      type:
+        type: string
+        enum:
+          - public_transport
+          - taxi_and_tnc
+          - shared_vehicles
```

Comment:
What’s the purpose of this?

Reply:
Is rather not introduce a new concept of mode groups beyond the mode identifiers, unless there’s a good reason too.

It also says that this field is required but it’s not a complete list of mode groups either.

Reply:
ok, so, you suggest to remove this?

Reply:
Yes, at least until we have a use case where we need it.